### PR TITLE
error if cluster in removing state

### PR DESCRIPTION
**Pull request by kinarashah**

### CODE REVIEW

File: controller/nodesyncer/nodessyncer.go [hunk 1]
```diff
@@ -86,15 +86,24 @@ func (n *NodeSyncer) createOrUpdateClusterNode(node *v1.Node) error {
 	clusterNode := n.convertNodeToClusterNode(node)
 	if existing == nil {
 		logrus.Infof("Creating cluster node [%s]", clusterNode.Name)
+		clusterNode.ObjectMeta.OwnerReferences = append(clusterNode.ObjectMeta.OwnerReferences, metav1.OwnerReference{Name: n.clusterName})
```
Comment (alena1108) [inline controller/nodesyncer/nodessyncer.go]: We should check the cluster state for both create and update

File: controller/nodesyncer/nodessyncer.go [hunk 2]
```diff
@@ -86,15 +86,24 @@ func (n *NodeSyncer) createOrUpdateClusterNode(node *v1.Node) error {
...
 		_, err := n.client.ClusterControllerClientV1.ClusterNodes("").Create(clusterNode)
 		if err != nil {
 			return fmt.Errorf("Failed to create cluster node [%s] %v", clusterNode.Name, err)
 		}
+		logrus.Infof("Created cluster node [%s]", clusterNode.Name)
 	} else {
+		cluster, err := n.client.ClusterControllerClientV1.Clusters("").Get(clusterNode.ObjectMeta.ClusterName, metav1.GetOptions{})
```
Comment (alena1108) [inline controller/nodesyncer/nodessyncer.go]: so this block should be moved up, outside of if/else

File: controller/nodesyncer/nodessyncer.go [hunk 3]
```diff
@@ -119,3 +124,14 @@ func (n *NodeSyncer) convertNodeToClusterNode(node *v1.Node) *clusterv1.ClusterN
 	}
 	return clusterNode
 }
+
+func (n *NodeSyncer) isClusterRemoving(clusterNode *clusterv1.ClusterNode) (bool, error) {
+	cluster, err := n.client.ClusterControllerClientV1.Clusters("").Get(clusterNode.ObjectMeta.ClusterName, metav1.GetOptions{})
+	if err != nil {
```
Comment (alena1108) [inline controller/nodesyncer/nodessyncer.go]: we should return true only when error is 404 not found

File: controller/nodesyncer/nodessyncer.go [hunk 4]
```diff
@@ -119,3 +124,14 @@ func (n *NodeSyncer) convertNodeToClusterNode(node *v1.Node) *clusterv1.ClusterN
...
+		return true, fmt.Errorf("Failed to get cluster [%s]", clusterNode.ObjectMeta.ClusterName)
+	}
+	if cluster.ObjectMeta.DeletionTimestamp != nil {
+		return true, fmt.Errorf("Cluster [%s] in removing state", cluster.Name)
```
Comment (alena1108) [inline controller/nodesyncer/nodessyncer.go]: error should be nil here

File: controller/nodesyncer/nodessyncer.go [hunk 5]
```diff
@@ -119,3 +124,14 @@ func (n *NodeSyncer) convertNodeToClusterNode(node *v1.Node) *clusterv1.ClusterN
 	}
 	return clusterNode
 }
+
+func (n *NodeSyncer) isClusterRemoving(clusterNode *clusterv1.ClusterNode) (bool, error) {
```
Comment (alena1108) [inline controller/nodesyncer/nodessyncer.go]: So here is the deal :) this method should return error only when k8s api call errors out with non-404 error. When there is an error and 404, return true, nil. When non error and removing timestamp is set, return true, nil. In all other cases, return false, nil.

**Comment by alena1108** [comment]
LGTM

**Comment by ibuildthecloud** [comment]
rebased in #9

**Pull request closed without merging**